content/en/docs/instrumentation/net/resources.md

```diff
@@ -0,0 +1,148 @@
+For example, a process producing telemetry that is running in a container on
+Kubernetes has a Pod name, a namespace, and possibly a deployment name. All
+three of these attributes can be included in the resource.
+
+In your observability backend, you can use resource information to better
+investigate interesting behavior. For example, if your trace or metrics data
+indicate latency in your system, you can narrow it down to a specific container,
+pod, or kubernetes deployment.
+
+## Setup
+
+Follow the instructions in the [Getting Started][], so that you have a running
+.NET app exporting data to the console.
+
+## Adding resources with environment variables
+
+You can use the `OTEL_RESOURCE_ATTRIBUTES` environment variable to inject
+resources into your application. The .NET SDK will automatically detect these
+resrouces.
+
+The following example adds [Service][], [Host][] and [OS][] resource attributes
+via environment variables, running unix programs like `uname` to generate the
+resource data.
+
+```console
+env OTEL_RESOURCE_ATTRIBUTES="service.name=resource-tutorial-dotnet,service.namespace=tutorial,service.version=1.0,service.instance.id=`uuidgen`,host.name=`HOSTNAME`,host.type=`uname -m`,os.name=`uname -s`,os.version=`uname -r`" dotnet run
+
+Activity.TraceId:          d1cbb7787440cc95b325835cb2ff8018
+Activity.SpanId:           2ca007300fcb3068
+Activity.TraceFlags:           Recorded
+Activity.ActivitySourceName: tutorial-dotnet
+Activity.DisplayName: SayHello
+Activity.Kind:        Internal
+Activity.StartTime:   2022-10-02T13:31:12.0175090Z
+Activity.Duration:    00:00:00.0003920
+Activity.Tags:
+    foo: 1
+    bar: Hello, World!
+    baz: [1,2,3]
+Resource associated with Activity:
+    service.name: resource-tutorial-dotnet
+    service.namespace: tutorial
+    service.version: 1.0
+    service.instance.id: 93B14BAD-813D-48EE-9FB1-2ADFD07C5E78
+    host.name: myhost
+    host.type: arm64
+    os.name: Darwin
+    os.version: 21.6.0
+```
+
+## Adding resources in code
+
+You can also add custom resources in code by attaching them to a
+`ResourceBuilder`.
+
+The following example builds on the [getting started] sample and adds two custom
+resources, `environment.name` and `team.name` in code:
+
+```csharp
+using System.Diagnostics;
+using System.Collections.Generic;
+
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Resources;
+
+var resourceBuilder =
+    ResourceBuilder
+        .CreateDefault()
+        .AddAttributes(new Dictionary<string, object>
+        {
+            ["environment.name"] = "production",
+            ["team.name"] = "backend"
+        });
+
+var sourceName = "tutorial-dotnet";
+
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource(sourceName)
+    .SetResourceBuilder(resourceBuilder)
+    .AddConsoleExporter()
+    .Build();
+
+var MyActivitySource = new ActivitySource(sourceName);
+
+using var activity = MyActivitySource.StartActivity("SayHello");
```

Comment:
I think that we should promote API where the Kind is specified. It is based on a bug we have had in AutoInstrumentation for .NET.

```suggestion
using var activity = MyActivitySource.StartActivity("SayHello", ActivityKind.Internal);
```

Reply:
Hmmm. Is this not the default though? So long as it's possible to specify a different kind of `ActivityKind` via .NET APIs then I'm not sure I'd agree with having this overload (specifying something that's a default) for every call to `StartActivity` in the docs.

Reply:
Internal is the default, so no need of explcitily specifying it.